### ISSUE TITLE
Add Hawaii Complainant Types other than Person in Custody

### ISSUE THREAD

**Issue by andrew-rist-tw** [PublicDataWorks/complaint-manager#246]
~Blocked by #228~

### Business Context:
We need to let the radio buttons for the Complainant Type (currently Civilian, Police Officer, Civilian (NOPD)) be determined on an org by org basis.  Since PERSON_TYPE hasn't been moved out of instance-files, we will go ahead and use it from instance-files, but we need to add a dimension to PERSON_TYPE that tells us whether there should be an input form (as civilian currently has) or whether the creation flow takes you to a new page to search for someone from a roster (as the other two types currently have).

**As a** new org adopting CM
**I want** to be able to use a dropdown for complainant type if I have more than four person types when creating a case and what happens when they're selected
**So that** I can attach complainant types to a case that make sense for my use-case

### In Scope
1. Add PERSON_TYPEs for the complainant types other than Person in custody
2. Toggle between radio buttons and dropdown select based on how many PERSON_TYPEs there are
3. Add a field on PERSON_TYPE that determines whether a sub-dropdown is show when a certain person type is selected and what values there will be in the dropdown

### Out of Scope
1. Adding people from the CaseDetailsPage

### Open Questions
1.
2.
4.

## Acceptance Criteria

1. Scenario: Department Staff
Given I have clicked the Create New Case button and the dialog has popped up 
When I select Complainant is a.... Department Staff
Then the "staff type" dropdown appears
AND I am able to select from that dropdown
AND THEN fill out their contact information

2. Scenario: Other ("Program Staff", "FamilyFriend", "Community Member/Organization", "Commissioner", "Legislator", see list from andrew in comment below)
Given I have clicked the Create New Case button and the dialog has popped up 
When I select one of the above options in Complainant Type
Then I am able to add their information (Name and some form of contact) (keep the required fields the same, can update later)

### Mockups

![](https://user-images.githubusercontent.com/86383902/214368599-9c85ec7a-b5fb-407b-acc1-8581c8124ca0.png)

![](https://user-images.githubusercontent.com/86383902/214368614-491ee850-bc04-43ec-b899-2a020884c363.png)

![](https://user-images.githubusercontent.com/86383902/214368654-2eafc4bc-1015-4430-81e2-0a8e067dc8e3.png)

### Checklist (should be reviewed before grooming as the BA or Pair are writing the card—NOT reviewed for the first time at grooming)
0. Discuss the technical implementation of this card.
1. Discuss how this feature would be included in or affect the Audit Log.
2. In respect to security, what could go wrong? Brainstorm threats using STRIDE model, especially if this card introduces a new technical architecture or design pattern. (TIMEBOX: 7 mins) https://docs.google.com/document/d/18AFl1d3SRieI4d_8R3GyEazqQOycLNDHGN_vOYIHuiE/edit#heading=h.aio9zb5j63wl
3. Ensure any open questions have been resolved with stakeholders.
5. Determine if documentation should be referenced or created.
9. Ensure to include all necessary mockups to this card.
10. Discuss what success/failure user feedback messages would be present, if any occur (front end/back end).
11. Discuss why this feature would not be implemented behind a toggle / what would happen if this card was partially done while a release branch was being cut.
12. If this card is part of a sequence of cards, discuss if this card should be blocked until the testing of previous  cards in the sequence have been finished.
13. Have we added edge cases as ACs that we're considered from a QA perspective
14. Does this card overlap with any entities outlined in [API Housekeeping](https://www.pivotaltracker.com/story/show/173963545)?  

**Notes:**

**Comment by andrew-rist-tw** [comment]
COMPLAINANT INFORMATION

* Remove:
  * Civilian
  * Police Officer
  * Civilian (NOPD)

* Add:
  * Department Staff - dropdown list (?) for:
    * Adult Correction Officer (ACO)
    * Medical Practitioner
    * Mental Health Professional
    * Program Staff
  * Person in Custody – Dropdown list (?) for Location:
    * Halawa Correctional Facility
    * Waiawa Correctional Facility
    * Kulani Correctional Facility
    * Saguaro Correctional Facility (private facility in AZ)
    * Oahu Community Correctional Center
    * Women’s Community Correctional Center
    * Hawaii Community Correctional Center
    * Kauai Community Correctional Center
    * Maui Community Correctional Center
* Family/Friend
* Community Member/Organization
* Legislator
* Commissioner
* Other

**Comment by andrew-rist-tw** [comment]
Hi there! We left some TODO comments where we couldn't complete a test (or in one case a non-test function) without having all of the Hawaii PersonTypes, so here's a list of files with those TODOs
- CreateCaseActions.test.js
- countComplaintsByComplainantType.test.js
- countMonthlyComplaintsByComplainantType.test.js
- getPersonType.js
- getPersonType.test.js
